File: benchmarks/suite5/fib.zig
const bench = @import("bench.zig");
const c = @cImport({
    @cInclude("stdio.h");
});

pub fn main() void {
    const n: i64 = 5000000;
    var a: i64 = 0;
    var b: i64 = 1;
    var i: i64 = 0;
    while (i < n) : (i += 1) {
        const nxt = a + b;
        a = b;
        b = nxt;
    }
    bench.printI64(a);
}

File: benchmarks/suite5/fib.c
#include <stdint.h>
#include <stdio.h>
#include "bench.h"

int main(void) {
  const int64_t n = 5000000;
  int64_t a = 0, b = 1;
  for (int64_t i = 0; i < n; i++) {
    int64_t c = a + b;
    a = b;
    b = c;
  }
  suite5_print_i64(a);
  return 0;
}
